frontend/src/components/nav.js

```diff
@@ -39,44 +39,56 @@ class NavigationBar extends Component {
   constructor(props) {
     super(props)
     this.state = {
-      isOpen: false,
+      isLead: false,
       loggedIn: false,
+      showLoginModal: false,
       username: null
     }
   }
 
   toggle = () => {
     this.setState({
-      isOpen: !this.state.isOpen
+      showLoginModal: !this.state.showLoginModal
     })
   }
+
   logout = () => {
     localStorage.removeItem('interviewerKey')
+    localStorage.removeItem('interviewerRoleIsLead')
+    localStorage.removeItem('interviewerName')
+
     this.setState({ loggedIn: false })
     alert('Logged Out!')
     Router.push('/')
   }
+
   async handleSubmit() {
     this.setState({
-      isOpen: false
+      showLoginModal: false
     })
     const { success, result } = await validateKey(this.state.currentKey)
     if (success) {
       localStorage.setItem('interviewerKey', this.state.currentKey)
+      localStorage.setItem('interviewerRoleIsLead', result.is_lead)
       localStorage.setItem('interviewerName', result.name)
       Router.push('/dashboard')
+
+      this.setState({
+        isLead: result.is_lead === 'true',
```

Comment:
Is this robust? It seems like I could just go to my browser console and type in `window.localStorage.setItem('interviewerRoleIsLead', true)` and manually set my own permissions level.

Reply:
No, this won't set your permission level. All this will do is show you the other nav links, but won't change your actual permission in terms of functionality.

Reply:
I think that's still bad practice, but probably not a big issue wrt the scope of what this portal does ¯\_(ツ)_/¯

Reply:
I agree that there's probably a better and more secure way to implement this - I will look into it. My intuition was that members won't edit local storage just to see pages in the nav that they won't have access to anyways.

Reply:
Changed implementation so it prevents local storage.